### PR TITLE
Updating GF180MCU DRC documentation

### LINKED ISSUE #13: Improve testing documentation to allow users to run locally
Improve testing documentation to allow users to run locally.

### PR DESCRIPTION
**Pull request by FaragElsayed2**
Fixes #13 

### Updating GF180MCU DRC documentation

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

**Comment by FaragElsayed2** [comment]
@atorkmabrains Could you please review this ?

File: klayout/drc/README.md [hunk 1]
```diff
@@ -2,44 +2,34 @@
...
+ ┣ 📜README.md                      This file to document the DRC run for GF180MCU.
+ ┗ 📜run_drc.py                     Main python script used for GF180MCU DRC.
  ```
 
-## Rule Deck Usage
-The `run_drc.py` script takes a gds file to run DRC rule decks of GF180 technology with switches to select subsets of all checks. 
+## **Prerequisites**
+You need the following set of tools installed to be able to run GF180MCU DRC:
+- Python 3.6+
+- KLayout 0.28.0+
```
Comment (atorkmabrains) [inline klayout/drc/README.md]: @FaragElsayed2 Please change the minimum klayout version to 0.28.4

**Review by atorkmabrains — CHANGES_REQUESTED** [review]

File: klayout/drc/testing/testcases/README.md [hunk 2]
```diff
@@ -1,7 +1,13 @@
-# DRC Unit Tests
+# Globalfoundries 180nm MCU DRC Unit Tests
```
Comment (atorkmabrains) [inline klayout/drc/testing/testcases/README.md]: @FaragElsayed2 Remove unit

File: klayout/drc/testing/testcases/README.md [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
 
-This folder has the following folders:
-* **switch_checking**      ( Contains a small test case to be used for testing the DRC switches. )
-* **torture**              ( Contains a few large test cases to test the performance of the rule deck. )
```
Comment (atorkmabrains) [inline klayout/drc/testing/testcases/README.md:5]: @FaragElsayed2 Why did you remove the switch_checking and torture testing?

**Review by atorkmabrains — CHANGES_REQUESTED** [review]

**Comment by atorkmabrains** [comment]
@proppy Please review.

File: klayout/drc/testing/testcases/README.md (hunk 2 above)
Comment (FaragElsayed2) [reply klayout/drc/testing/testcases/README.md]: Done

File: klayout/drc/README.md (hunk 1 above)
Comment (FaragElsayed2) [reply klayout/drc/README.md]: Done

Comment (FaragElsayed2) [reply klayout/drc/README.md]: Updated in run_drc & run_regression python scripts

File: klayout/drc/testing/testcases/README.md (hunk 3 above)
Comment (FaragElsayed2) [reply klayout/drc/testing/testcases/README.md:5]: Added

File: klayout/drc/README.md (hunk 1 above)
Comment (atorkmabrains) [reply klayout/drc/README.md]: Thanks @FaragElsayed2

**Review by atorkmabrains — APPROVED** [review]
Thanks @FaragElsayed2 for your work.

**Comment by atorkmabrains** [comment]
@proppy Could you please check and approve?

File: klayout/drc/README.md [hunk 4]
```diff
@@ -2,44 +2,34 @@
 
 Explains how to use the runset.
 
-## Folder Structure
+## **Folder Structure**
```
Comment (proppy) [inline klayout/drc/README.md]: I don't think we need to make this bold (header are already emphased when rendered in github)

File: klayout/drc/run_drc.py [hunk 5]
```diff
@@ -340,7 +340,13 @@ def check_klayout_version():
         exit(1)
     elif len(klayout_v_list) >= 2 and len(klayout_v_list) <= 3:
         if klayout_v_list[1] < 28:
-            logging.error("Prerequisites at a minimum: KLayout 0.28.0")
+            logging.error("Prerequisites at a minimum: KLayout 0.28.4")
+            logging.error(
+                "Using this klayout version has not been assesed in this development. Limits are unknown"
```
Comment (proppy) [inline klayout/drc/run_drc.py]: assessed?

can we simply say it's not supported?

File: klayout/drc/run_drc.py [hunk 6]
```diff
@@ -340,7 +340,13 @@ def check_klayout_version():
...
+            )
+            exit(1)
+        if klayout_v_list[1] == 28 and klayout_v_list[2] <= 3:
+            logging.error("Prerequisites at a minimum: KLayout 0.28.4")
```
Comment (proppy) [inline klayout/drc/run_drc.py]: can't we combine the two checks?

File: klayout/drc/testing/README.md [hunk 7]
```diff
@@ -1,22 +1,70 @@
...
+    ┣ 📜 <table_name>.drc                     
+    ┣ 📜 <table_name>_main_analysis.drc  
+    ┣ 📜 <table_name>_main.lyrdb        
+    ┣ 📜 <table_name>_main_markers_merged_final.lyrdb
+    ┣ 📜 <table_name>_main_markers.gds  
+    ┣ 📜 <table_name>_main_markers_merged.gds
+ ```
+
+The result is a database file (`<table_name>_main_markers_merged_final.lyrdb`) contains all violations. 
+You could view it on your file using: `klayout <table_name>_main_markers_merged.gds -m <table_name>_main_markers_merged_final.lyrdb`, or you could view it via makrer browser option in tools menu using klayout GUI.
```
Comment (proppy) [inline klayout/drc/testing/README.md]: s/makrer/marker?

Comment (proppy) [inline klayout/drc/testing/README.md]: can we give the instruction to open this in the klayout gui?

File: klayout/drc/testing/run_regression.py [hunk 8]
```diff
@@ -122,11 +122,18 @@ def check_klayout_version():
         exit(1)
     elif len(klayout_v_list) >= 2 or len(klayout_v_list) <= 3:
         if klayout_v_list[1] < 28:
-            logging.error("Prerequisites at a minimum: KLayout 0.28.0")
+            logging.error("Prerequisites at a minimum: KLayout 0.28.4")
```
Comment (proppy) [inline klayout/drc/testing/run_regression.py:125]: wouldn't this be also done by `run_drc` already?

File: klayout/drc/README.md [hunk 9]
```diff
@@ -48,13 +38,19 @@ Example:
 
 - `--path=<file_path>`                  The input GDS file path.
 
-- `--gf180mcu=<combined_options>`       Select combined options of metal_top, mim_option, and metal_level. Allowed values (A, B, C).
+- `--variant=<combined_options>`        Select combined options of metal_top, mim_option, and metal_level. Allowed values (A, B, C).
```
Comment (proppy) [inline klayout/drc/README.md:41]: I don't see the corresponding code change, was the doc incorrect in the first place?

Comment (atorkmabrains) [reply klayout/drc/README.md:41]: @proppy Yes, the doc was old and we added it as a place holder.

File: klayout/drc/testing/run_regression.py (hunk 8 above)
Comment (FaragElsayed2) [reply klayout/drc/testing/run_regression.py:125]: Yes, but we need it also in the regression script to raise this error before starting in regression testing steps. As there are some other steps before calling run_drc script.

File: klayout/drc/README.md (hunk 4 above)
Comment (FaragElsayed2) [reply klayout/drc/README.md]: Done

File: klayout/drc/run_drc.py (hunk 5 above)
Comment (FaragElsayed2) [reply klayout/drc/run_drc.py]: Done

File: klayout/drc/testing/README.md (hunk 7 above)
Comment (FaragElsayed2) [reply klayout/drc/testing/README.md]: Done

Comment (FaragElsayed2) [reply klayout/drc/testing/README.md]: Is that okay ?

File: klayout/drc/run_drc.py (hunk 6 above)
Comment (FaragElsayed2) [reply klayout/drc/run_drc.py]: Is that okay ?

Comment (atorkmabrains) [reply klayout/drc/run_drc.py]: @proppy I think we have agreed that those changes will be addressed later in #6 about moving this common functionality in some other place.

Could you please focus in this PR on documentation changes? We have made some minor changes in the code to match the version number required for klayout.

File: klayout/drc/testing/run_regression.py (hunk 8 above)
Comment (atorkmabrains) [reply klayout/drc/testing/run_regression.py:125]: @proppy Same comment at as above #6

File: klayout/drc/run_drc.py (hunk 6 above)
Comment (proppy) [reply klayout/drc/run_drc.py]: @atorkmabrains I agree with you (about focusing the PR on documentation), but this PR is adding a new version check (line 348 to 351); therefore I think it makes sense to ask why it is necessary and why it can't be achieved by changing the condition on line 341.

File: klayout/drc/testing/run_regression.py (hunk 8 above)
Comment (proppy) [reply klayout/drc/testing/run_regression.py:125]: Same comment here, this PR is adding a new version check on line 130-137, so it makes sense to review it here.

File: klayout/drc/run_drc.py (hunk 6 above)
Comment (atorkmabrains) [reply klayout/drc/run_drc.py]: @FaragElsayed2 Could you please make all the changes @proppy needs?

Comment (FaragElsayed2) [reply klayout/drc/run_drc.py]: Done

File: klayout/drc/testing/run_regression.py (hunk 8 above)
Comment (FaragElsayed2) [reply klayout/drc/testing/run_regression.py:125]: Done

**Review by proppy — APPROVED** [review]

**Comment by proppy** [comment]
Thanks for your work on reconcilling the repositories!
